src/DataStructures/Index.hpp

```diff
@@ -110,6 +123,16 @@ class Index {
   const std::array<size_t, Dim>& indices() const noexcept { return indices_; }
 
  private:
+  template <size_t N>
+  // NOLINTNEXTLINE(readability-redundant-declaration)
+  friend size_t collapsed_index(const Index<N>& index,
```

Comment:
Why does this need to be a `friend`? Looks like it only calls the public member function and could call `indices()` instead of using the private attribute